### DISCUSSION TITLE
Update of Docker images inside a GitHub action workflow

### DISCUSSION THREAD

**Discussion by Xendar** [renovatebot/renovate#15305] [Request Help] [answered]
Hello everyone,

I am using a GitHub action to run Renovate for my organization, under the form of invoking the Renovate Docker image (for some peculiar reasons I will omit here, we can't use the NPM CLI or the GitHub App).

This is basically my workflow (some details are omitted for brevity):

```yaml
name: Renovate

on:
  workflow_dispatch:
  schedule:
    # The "*" (#42, asterisk) character has special semantics in YAML, so this string has to be quoted.
    - cron: '0 */12 * * 1-5'

jobs:
  renovate:
    name: Renovate
    runs-on: [ self-hosted, linux ]
    steps:
      - name: Checkout
        uses: actions/checkout@dcd71f646680f2efd8db4afa5ad64fdcba30e748 # tag=v3.0.1

      - name: Run Renovate
        run: |
          docker run --rm \
            --volume ${{ github.workspace }}/renovate-config.js:/usr/src/app/config.js \
            --env LOG_LEVEL=debug \
            mycompany.net/my-registry-remote/renovate/renovate:32.26.3
```

This works fine.
Now I want Renovate to renovate as well this file, and in consequence the Renovate image.
As the `github-actions` manager [supports docker only under the form](https://github.com/renovatebot/renovate/blob/main/lib/modules/manager/github-actions/extract.ts#L8) of `uses: docker://` (by the way it would be worth having this documented :)) I started first by defining a regex:

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "local>my-org/renovate-config"
  ],
  "regexManagers": [
    {
      "fileMatch": [
        "renovate.yml"
      ],
      "matchStrings": [
        "mycompany.net/my-registry-remote/(?<depName>.*?):(?<currentValue>.*?)\\s"
      ],
      "datasourceTemplate": "docker"
    }
  ]
}
```

This works as well. However, it does not support the `pinDigests` feature which is enabled in my `renovate-config`...
I tried with the `kubenetes` manager as a hack, but this won't work as the manager is asserting if the file is indeed a [K8S manifest](https://github.com/renovatebot/renovate/blob/main/lib/modules/manager/kubernetes/extract.ts#L10-L15).

So long story short, is there a way to be able to get Renovate to match the Docker image **AND** to get the pin digests feature?

Thanks!

**Comment by viceice** [comment, marked as answer]
you need to manually pin the docker images and add `currentDigest` to match regex.

btw: we also have a official GitHub action

https://github.com/renovatebot/github-action

**Reply by Xendar** [reply]
> you need to manually pin the docker images and add `currentDigest` to match regex.
> 
> btw: we also have a official GitHub action
> 
> https://github.com/renovatebot/github-action

I totally missed the `currentDigest` option in the documentation 🤯
Thanks @viceice I'll check that and provide the working updated sample!

**Reply by Xendar** [reply]
I can confirm indeed that using the `currentDigest` fixes my issue:

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "local>my-org/renovate-config"
  ],
  "regexManagers": [
    {
      "fileMatch": [
        "renovate.yml"
      ],
      "matchStrings": [
        "mycompany.net/my-registry/(?<depName>.*?):(?<currentValue>.*?)@(?<currentDigest>.*?)\\s"
      ],
      "datasourceTemplate": "docker"
    }
  ]
}
```

Thanks again @viceice !
